### PR TITLE
371 add function to know if gripper has finished movement

### PR DESCRIPTION
**Pull request by glannuzel**
We have to find a name close to `is_move_finished()` but which does not confuse the user that moving the gripper is not a move returning an id etc...

### CODE REVIEW

File: src/reachy2_sdk/parts/hand.py [hunk 1]
```diff
@@ -78,6 +86,8 @@ def set_opening(self, percentage: float) -> None:
                 ),
             )
         )
+        self._request_sent = True
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/hand.py:89]: don't we need this in the send_goal_positions also?

File: src/reachy2_sdk/parts/hand.py [hunk 2]
```diff
@@ -126,12 +140,29 @@ def is_off(self) -> bool:
...
+
     def _update_with(self, new_state: HandState) -> None:
         """Update the hand with a newly received (partial) state received from the gRPC server."""
         self._present_position = new_state.present_position.parallel_gripper.position.value
         self._goal_position = new_state.goal_position.parallel_gripper.position.value
         self._opening = new_state.opening.value
         self._compliant = new_state.compliant.value
+        if self._request_sent:
+            if len(self._last_present_positions) < self._last_present_positions_queue_size:
+                pass
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/hand.py:158]: just a pass here is weird. The if and elif could be combined into on condition. what do you think?

File: src/reachy2_sdk/parts/hand.py [hunk 3]
```diff
@@ -126,12 +140,29 @@ def is_off(self) -> bool:
...
+            elif np.isclose(self._present_position, self._last_present_positions[-1], np.deg2rad(0.1)) and np.isclose(
```
Comment (FabienDanieau) [inline src/reachy2_sdk/parts/hand.py:159]: I'd make a simple function called check gripper movement (something like that). I'd be easier to understand what we are testing here.

File: src/reachy2_sdk/parts/hand.py [hunk 4]
```diff
@@ -126,12 +140,29 @@ def is_off(self) -> bool:
         """Get compliancy of the hand"""
         return self._compliant
 
+    def is_motion_complete(self) -> bool:
```
[the remark this answers is not in the document]
Comment (glannuzel) [reply src/reachy2_sdk/parts/hand.py:143]: Changed into `is_moving()`, so no confusion will be possible

File: tests/units/online/test_hand.py [hunk 5]
```diff
@@ -67,3 +67,25 @@ def test_gripper_off(reachy_sdk_zeroed: ReachySDK) -> None:
...
+    assert reachy_sdk_zeroed.r_arm.gripper.is_motion_complete()
+    assert np.isclose(reachy_sdk_zeroed.r_arm.gripper.opening, 100, 1e-01)
+
+    reachy_sdk_zeroed.r_arm.gripper.close()
+    assert not reachy_sdk_zeroed.r_arm.gripper.is_motion_complete()
+    time.sleep(1.0)
+    assert reachy_sdk_zeroed.r_arm.gripper.is_motion_complete()
+    assert np.isclose(reachy_sdk_zeroed.r_arm.gripper.opening, 0, 1e-01)
+
+    reachy_sdk_zeroed.r_arm.gripper.open()
```
[the remark this answers is not in the document]
Comment (glannuzel) [reply tests/units/online/test_hand.py:87]: Removed, I modified the conftest to be control the initial state

File: tests/units/online/test_hand.py [hunk 6]
```diff
@@ -67,3 +67,25 @@ def test_gripper_off(reachy_sdk_zeroed: ReachySDK) -> None:
     time.sleep(3.0)
     assert reachy_sdk_zeroed.l_arm.is_on() == True
     assert np.isclose(reachy_sdk_zeroed.l_arm.elbow.pitch.present_position, -60, 10)
+
+
+@pytest.mark.online
+def test_gripper_motion_complete(reachy_sdk_zeroed: ReachySDK) -> None:
```
[the remark this answers is not in the document]
Comment (glannuzel) [reply tests/units/online/test_hand.py:73]: Done with left_gripper too

[the remark this answers is not in the document]
Comment (glannuzel) [reply tests/units/online/test_hand.py:73]: Done

File: src/reachy2_sdk/parts/hand.py (hunk 3 above)
Comment (glannuzel) [reply src/reachy2_sdk/parts/hand.py]: Done

File: src/reachy2_sdk/parts/hand.py (hunk 2 above)
Comment (glannuzel) [reply src/reachy2_sdk/parts/hand.py]: Done

**Review by ClaireHzl — APPROVED** [review]
works perfectly ! :)

**Pull request merged**